operator/controllers/resources/utils.go

```diff
@@ -0,0 +1,58 @@
+package resources
+
+import (
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/trace"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	ConditionReasonOriginLabelNotMatching = "OriginLabelNotMatching"
+	ConditionReasonOriginLabelMatching    = "OriginLabelMatching"
+	ConditionReasonNewResource            = "NewResource"
+	ConditionTypeTeleportResourceOwned    = "TeleportResourceOwned"
+)
+
+// isResourceOriginKubernetes reads a teleport resource metadata, searches for the origin label and checks its
+// value is kubernetes.
+func isResourceOriginKubernetes(resource types.Resource) bool {
+	metadata := resource.GetMetadata()
+	if label, ok := metadata.Labels[types.OriginLabel]; ok {
+		return label == types.OriginKubernetes
+	}
+	return false
+}
+
+// checkOwnership takes an existing resource and validates the operator owns it.
+// It returns an ownership condition and an error if the resource is not owned by the operator
+func checkOwnership(existingResource types.Resource) (metav1.Condition, error) {
+	if existingResource != nil {
+		if !isResourceOriginKubernetes(existingResource) {
```

Comment:
:nail_care: 
I think we could decrease this function's max indentation by checking first if `existingResource == nil`